### PR TITLE
net/sockif: add si_getsockopt and si_setsockopt to simply getsockopt and setsockopt

### PR DESCRIPTION
**Pull request by Donny9**
## Summary
1. net/sockif: add si_getsockopt and si_setsockopt to simply getsockopt and setsockopt
2. support SO_PEERCRED for local socket.
## Impact
simply code
## Testing
local test

### CODE REVIEW

File: include/sys/socket.h [hunk 1]
```diff
@@ -330,6 +333,13 @@ struct cmsghdr
   int cmsg_type;                /* Protocol-specific type */
 };
 
+struct ucred
+{
+  pid_t pid;
+  pid_t uid;
+  pid_t gid;
```
Comment (pkarashchenko) [inline include/sys/socket.h:340]:
```suggestion
  uid_t uid;
  gid_t gid;
```

Comment (Donny9) [reply include/sys/socket.h:340]: Done! Thanks.

**Review by pkarashchenko — APPROVED** [review]

File: net/local/local_sockif.c [hunk 2]
```diff
@@ -94,6 +100,10 @@ const struct sock_intf_s g_local_sockif =
   local_close,       /* si_close */
   local_ioctl,       /* si_ioctl */
   local_socketpair   /* si_socketpair */
+#ifdef CONFIG_NET_SOCKOPTS
+  local_getsockopt,  /* si_getsockopt */
+  local_setsockopt,  /* si_setsockopt */
```
Comment (pkarashchenko) [inline net/local/local_sockif.c:105]:
```suggestion
  , local_getsockopt /* si_getsockopt */
  , local_setsockopt /* si_setsockopt */
```
here and other similar places

**Review by xiaoxiang781216 — APPROVED** [review]

**Comment by masayuki2009** [comment]
@Donny9 @xiaoxiang781216 
Hmm, telnetd does not work with spresense:wifi now.

**Comment by masayuki2009** [comment]
@Donny9 @xiaoxiang781216 

`sim:usrsocktest` also failed.

```
NuttShell (NSH) NuttX-10.4.0
nsh> uname -a
NuttX  10.4.0 ccf0c3566e Nov 23 2022 17:30:48 sim sim
nsh> usrsocktest
Starting unit-tests...
Testing group "char_dev" =>
	Group "char_dev": [OK]
Testing group "no_daemon" =>
	Group "no_daemon": [OK]
Testing group "basic_daemon" =>
	Group "basic_daemon": [OK]
Testing group "basic_connect" =>
	Group "basic_connect": [OK]
Testing group "basic_connect_delay" =>
	Group "basic_connect_delay": [OK]
Testing group "no_block_connect" =>
	Group "no_block_connect": [OK]
Testing group "basic_send" =>
	Group "basic_send": [OK]
Testing group "no_block_send" =>
	Group "no_block_send": [OK]
Testing group "block_send" =>
	Group "block_send": [OK]
Testing group "no_block_recv" =>
	Group "no_block_recv": [OK]
Testing group "block_recv" =>
	Group "block_recv": [OK]
Testing group "remote_disconnect" =>
	Group "remote_disconnect": [OK]
Testing group "basic_setsockopt" =>
	[TEST ASSERT FAILED!]
		In function "basic_setsockopt_open":
		line 103: Assertion `(ssize_t)((ret)) == (ssize_t)((0))' failed.
			got value: -1
			should be: 0
	Group "basic_setsockopt": [FAILED]
Testing group "basic_getsockopt" =>
	[TEST ASSERT FAILED!]
		In function "basic_getsockopt_open":
		line 89: Assertion `(ssize_t)((usrsocktest_daemon_start(dconf))) == (ssize_t)((OK))' failed.
			got value: -114
			should be: 0
	Group "basic_getsockopt": [FAILED]
Testing group "basic_getsockname" =>
	[TEST ASSERT FAILED!]
		In function "basic_getsockname_open":
		line 94: Assertion `(ssize_t)((usrsocktest_daemon_start(dconf))) == (ssize_t)((OK))' failed.
			got value: -114
			should be: 0
	Group "basic_getsockname": [FAILED]
Testing group "wake_with_signal" =>
	[TEST ASSERT FAILED!]
		In function "do_wake_test":
		line 517: Assertion `(ssize_t)((usrsocktest_daemon_start(&usrsocktest_daemon_config))) == (ssize_t)((OK))' failed.
			got value: -114
			should be: 0
	Group "wake_with_signal": [FAILED]
Testing group "multithread" =>
	[TEST ASSERT FAILED!]
		In function "usrsocktest_test_multithread_open_close":
		line 210: Assertion `(ssize_t)((usrsocktest_daemon_start(&usrsocktest_daemon_config))) == (ssize_t)((OK))' failed.
			got value: -114
			should be: 0
	Group "multithread": [FAILED]
Unit-test groups done... OK:12, FAILED:5, TOTAL:17
 -- number of checks made: 2522
```

**Comment by masayuki2009** [comment]
@xiaoxiang781216 
May I revert this PR until the issue has been fixed?

**Comment by jerpelea** [comment]
+1 for revert

**Comment by xiaoxiang781216** [comment]
@Donny9 will take a look in the next couple month, I am interesting why citest report usrsocktest pass here:
```
https://github.com/apache/incubator-nuttx/actions/runs/3529252426/jobs/5920152512
test_example/test_example.py::test_hello PASSED                          [  8%]
test_example/test_example.py::test_helloxx PASSED                        [ 16%]
test_example/test_example.py::test_pipe PASSED                           [ 25%]
test_example/test_example.py::test_popen PASSED                          [ 33%]
test_example/test_example.py::test_usrsocktest PASSED                    [ 41%]
test_os/test_os.py::test_ostest PASSED                                   [ 50%]
test_os/test_os.py::test_mm PASSED                                       [ 58%]
test_os/test_os.py::test_cxxtest PASSED                                  [ 66%]
test_os/test_os.py::test_scanftest PASSED                                [ 75%]
test_os/test_os.py::test_getprime PASSED                                 [ 83%]
test_os/test_os.py::test_fs_test PASSED                                  [ 91%]
test_os/test_os.py::test_psram_test PASSED                               [100%]
```
@ttnie could you take a look?

**Pull request merged**